source/js/iframeAcceptance.js

```diff
@@ -1,68 +1,75 @@
-        .forEach(iframeWrapper => {
-            const iframe = iframeWrapper.querySelector('iframe');
-            const iframeUrl = new URL(iframe.getAttribute('data-src'));
-            const iframeWrapperParent = iframeWrapper.parentNode;
-            if(acceptedSuppliers.includes(iframeUrl.host)) {
-                if(iframeWrapperParent.classList.contains('embed__ratio--1-1')) {
-                    iframeWrapperParent.classList.replace('embed__ratio--1-1', 'embed__ratio--16-9');
-                 }
-                iframe.setAttribute('src', iframe.getAttribute('data-src'));
-                iframeWrapper.classList.remove('js-suppressed-iframe-wrapper');
-                iframeWrapper.style.position = 'static';
-                iframeWrapper.querySelector('.js-suppressed-iframe-prompt').classList.add('u-display--none');
-            } 
-        }); 
-}
+/* Sets local storage */
+const setLocalStorage = (contentWrapper) => {
+    const iframeUrl = new URL(contentWrapper.getAttribute('data-src'));
 
-const onClicklHandler = (iframeWrapper) => {
-    const iframeUrl = new URL(iframeWrapper.querySelector('iframe').getAttribute('data-src'));
     if (!acceptedSuppliers.includes(iframeUrl.host) && iframeUrl.host !== "https" && iframeUrl.host !== "http") {
         acceptedSuppliers.push(iframeUrl.host);
-    } 
-    
+    }
     localStorage.setItem('acceptedSuppliers', JSON.stringify(acceptedSuppliers));
-    
-    revealIframes();
 }
 
-const suppressIframes = () => {
-    [...document.querySelectorAll('.js-suppressed-iframe-wrapper')]
-    .forEach(iframeWrapper => {
-            const buttonEl = iframeWrapper.querySelector('[js-suppressed-iframe-button]');
-            buttonEl.params = {iframe: iframeWrapper};
-            buttonEl.addEventListener('click', () => {
-                onClicklHandler(iframeWrapper);
-            });
-           
-        });
+/* Reveal function */
+const revealContent = (contentWrapper) => {
+    const template = contentWrapper.querySelector('template');
+    const suppressedContentWrapper = contentWrapper.querySelector('.c-acceptance__content');
+    const clone = template.content.cloneNode(true);
+    suppressedContentWrapper.appendChild(clone);
+    const suppressedContent = template.nextElementSibling;
+    contentWrapper.classList.remove('js-suppressed-content');
+    contentWrapper.querySelector('.js-suppressed-content-prompt').classList.add('u-display--none');
+    suppressedContent.setAttribute('src', suppressedContent.getAttribute('data-src'));
 }
 
-export default () => addEventListener('DOMContentLoaded', () => {
-    if (window.innerHeight > window.innerWidth) {
-        [...document.querySelectorAll('.embed__ratio--16-9')].forEach(embed => {
-            embed.classList.replace('embed__ratio--16-9', 'embed__ratio--1-1');
-        });
+/* Loops through an reveal every URL-host matching local storage */
+const revealContentLoop = () => {
+    [...document.querySelectorAll('.js-suppressed-content')]
```

Comment:
What if the class .js-suppressed-content dosent exist? There are no undefined check here.